tests/test_downloadermiddleware_httpcompression.py

```diff
@@ -249,3 +262,34 @@ def test_process_response_head_request_no_decode_required(self):
         newresponse = self.mw.process_response(request, response, self.spider)
         self.assertIs(newresponse, response)
         self.assertEqual(response.body, b'')
+
+    def test_process_response_gzip_keep_headers(self):
+        self.spider, self.mw = self.create_spider_mw(
```

Comment:
```suggestion
        spider, mw = self.create_spider_mw(
```

Don’t overwrite the existing class variables. Simply declare new ones local to the test function, and use those instead of the `self.*` equivalent in the rest of the test method. Same for the other new test method.

Reply:
Would it better or indifferent if I assigned them names such as `test_spider` and `test_mw` to not have them get confused with the self variables?